src/NationalInstruments.Analyzers/Style/LongChainDottedInvocationAnalyzer.cs

```diff
@@ -0,0 +1,115 @@
+            new LocalizableResourceString(nameof(Resources.NI1017_Title), Resources.ResourceManager, typeof(Resources)),
+            new LocalizableResourceString(nameof(Resources.NI1017_Message), Resources.ResourceManager, typeof(Resources)),
+            Resources.CategoryNI,
+            DiagnosticSeverity.Warning,
+            isEnabledByDefault: true);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecutionIf(IsRunningInProduction);
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+            context.RegisterSyntaxNodeAction(AnalyzeExpressionStatementSyntax, SyntaxKind.ExpressionStatement);
+            context.RegisterSyntaxNodeAction(AnalyzeEqualsValueClauseSyntax, SyntaxKind.EqualsValueClause);
+            context.RegisterSyntaxNodeAction(AnalyzeArgumentSyntax, SyntaxKind.Argument);
+            context.RegisterSyntaxNodeAction(AnalyzeArrowExpressionClauseSyntax, SyntaxKind.ArrowExpressionClause);
+        }
+
+        private static void AnalyzeExpressionStatementSyntax(SyntaxNodeAnalysisContext context)
+        {
+            var expressionStatementSyntax = (ExpressionStatementSyntax)context.Node;
+
+            AnalyzeSyntax(expressionStatementSyntax, context);
+        }
+
+        private static void AnalyzeEqualsValueClauseSyntax(SyntaxNodeAnalysisContext context)
+        {
+            var equalsValueClauseSyntax = (EqualsValueClauseSyntax)context.Node;
+
+            AnalyzeSyntax(equalsValueClauseSyntax, context);
+        }
+
+        private static void AnalyzeArrowExpressionClauseSyntax(SyntaxNodeAnalysisContext context)
+        {
+            var expressionStatementSyntax = (ArrowExpressionClauseSyntax)context.Node;
+
+            AnalyzeSyntax(expressionStatementSyntax, context);
+        }
+
+        private static void AnalyzeArgumentSyntax(SyntaxNodeAnalysisContext context)
+        {
+            var argumentSyntax = (ArgumentSyntax)context.Node;
+
+            AnalyzeSyntax(argumentSyntax, context);
+        }
+
+        private static void AnalyzeSyntax(SyntaxNode parentSyntaxNode, SyntaxNodeAnalysisContext context)
+        {
+            // Find the invocation expression i.e
+            // a method/delegate call or a property access
+            // or a chain of them
+            var invocationExpressionSyntax = parentSyntaxNode
+                .DescendantNodes()
+                .OfType<InvocationExpressionSyntax>()
+                .FirstOrDefault();
+
+            if (invocationExpressionSyntax is null)
+            {
+                // This does not contain any invocations, bail out
+                return;
+            }
+
+            // Find all invocations which involve brackets ()
+            var nonNestedInvocationsWithParenthesis = invocationExpressionSyntax
```

Comment:
How do we know these aren't nested? Perhaps switch from DescendantNodes to ChildNodes?

Reply:
> How do we know these aren't nested?

The lambda `syntaxNode => !syntaxNode.IsKind(SyntaxKind.Argument)` in `DescendantNodes()` ensures that we do not get any of the nested invocations. The `DescendantNodes()` call does not go inside any parenthesis (`SyntaxKind.Argument`).

> Perhaps switch from DescendantNodes to ChildNodes?

I tried this actually. But, could not get it to work. See the syntax tree - 

![image](https://user-images.githubusercontent.com/71984539/96257133-77f71980-0fd7-11eb-8048-77284cc51feb.png)

The `InvocationExpressionSyntax` are nested, so I needed to call `DescendantNodes()`. Do you have any other ideas?